[bug]: Docker Compose Backend container always restarting

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I have cloned the repository, copied the .env.example file to .env and changed with custom settings

Running docker compose up -ad the backend container is build but continous restarting with following error:

hoppscotch-backend  | node:internal/modules/cjs/loader:958
hoppscotch-backend  |   throw err;
hoppscotch-backend  |   ^
hoppscotch-backend  | 
hoppscotch-backend  | Error: Cannot find module '/usr/src/app/dist/main'
hoppscotch-backend  |     at Module._resolveFilename (node:internal/modules/cjs/loader:955:15)
hoppscotch-backend  |     at Module._load (node:internal/modules/cjs/loader:803:27)
hoppscotch-backend  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
hoppscotch-backend  |     at node:internal/main/run_main_module:18:47 {
hoppscotch-backend  |   code: 'MODULE_NOT_FOUND',
hoppscotch-backend  |   requireStack: []
hoppscotch-backend  | }

### Steps to reproduce

Clone repository
Copy .env.example to .env
Execute docker compose up -d

### Environment

Production

### Version

Self-hosted

> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Current behavior
> I have cloned the repository, copied the .env.example file to .env and changed with custom settings
> 
> Running docker compose up -ad the backend container is build but continous restarting with following error:
> 
> hoppscotch-backend | node:internal/modules/cjs/loader:958 hoppscotch-backend | throw err; hoppscotch-backend | ^ hoppscotch-backend | hoppscotch-backend | Error: Cannot find module '/usr/src/app/dist/main' hoppscotch-backend | at Module._resolveFilename (node:internal/modules/cjs/loader:955:15) hoppscotch-backend | at Module._load (node:internal/modules/cjs/loader:803:27) hoppscotch-backend | at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) hoppscotch-backend | at node:internal/main/run_main_module:18:47 { hoppscotch-backend | code: 'MODULE_NOT_FOUND', hoppscotch-backend | requireStack: [] hoppscotch-backend | }
> 
> ### Steps to reproduce
> Clone repository Copy .env.example to .env Execute docker compose up -d
> 
> ### Environment
> Production
> 
> ### Version
> Self-hosted

my solution is comment the following line at docker-compose.yml, hope this can help you 
<img width="560" alt="image" src="https://github.com/hoppscotch/hoppscotch/assets/1164959/bee8b249-1d17-476b-9957-c157a0fbf1eb">
https://github.com/hoppscotch/hoppscotch/blob/d6c840011616317ba98f143da61b8f38bc1f5fc2/docker-compose.yml#L22

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -67,11 +67,13 @@ services:
       - "5432:5432"
     user: postgres
     environment:
+      # The default user defined by the docker image
+      POSTGRES_USER: postgres
       # NOTE: Please UPDATE THIS PASSWORD!
       POSTGRES_PASSWORD: testpass
       POSTGRES_DB: hoppscotch
     healthcheck:
-      test: [ "CMD-SHELL", "pg_isready" ]
+      test: ["CMD-SHELL", "sh -c 'pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB}'"]
       interval: 5s
       timeout: 5s
       retries: 10

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -78,8 +78,8 @@ services:
       POSTGRES_DB: hoppscotch
     healthcheck:
       test: [ "CMD-SHELL", "pg_isready" ]
-      interval: 10s
+      interval: 5s
       timeout: 5s
-      retries: 5
+      retries: 10
 
 

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -27,12 +27,6 @@ services:
         condition: service_healthy
     ports:
       - "3170:3000"
-    command:
-      - /bin/sh
-      - -c
-      - |
-        pnpm exec prisma migrate deploy
-        pnpm run start:prod
 
   # The main hoppscotch app. This will be hosted at port 3000
   # NOTE: To do TLS or play around with how the app is hosted, you can look into the Caddyfile for

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -19,6 +19,8 @@ services:
       - DATABASE_URL=postgresql://postgres:testpass@hoppscotch-db:5432/hoppscotch?connect_timeout=300
       - PORT=3000
     volumes:
+      # Uncomment the line below when modifying code. Only applicable when using the "dev" target.
+      # - ./packages/hoppscotch-backend/:/usr/src/app
       - /usr/src/app/node_modules/
     depends_on:
       hoppscotch-db:
@@ -66,7 +68,7 @@ services:
   # you are using an external postgres instance
   # This will be exposed at port 5432
   hoppscotch-db:
-    image: postgres:15.3-alpine3.18
+    image: postgres:15
     ports:
       - "5432:5432"
     user: postgres

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -19,12 +19,18 @@ services:
       - DATABASE_URL=postgresql://postgres:testpass@hoppscotch-db:5432/hoppscotch?connect_timeout=300
       - PORT=3000
     volumes:
-      - ./packages/hoppscotch-backend/:/usr/src/app
       - /usr/src/app/node_modules/
     depends_on:
-      - hoppscotch-db
+      hoppscotch-db:
+        condition: service_healthy
     ports:
       - "3170:3000"
+    command:
+      - /bin/sh
+      - -c
+      - |
+        pnpm exec prisma migrate deploy
+        pnpm run start:prod
 
   # The main hoppscotch app. This will be hosted at port 3000
   # NOTE: To do TLS or play around with how the app is hosted, you can look into the Caddyfile for
@@ -60,12 +66,18 @@ services:
   # you are using an external postgres instance
   # This will be exposed at port 5432
   hoppscotch-db:
-    image: postgres
+    image: postgres:15.3-alpine3.18
     ports:
       - "5432:5432"
+    user: postgres
     environment:
       # NOTE: Please UPDATE THIS PASSWORD!
       POSTGRES_PASSWORD: testpass
       POSTGRES_DB: hoppscotch
+    healthcheck:
+      test: [ "CMD-SHELL", "pg_isready" ]
+      interval: 10s
+      timeout: 5s
+      retries: 5
 
 

